components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/utils/FileBasedApplicationImportExportManager.java

```diff
@@ -0,0 +1,218 @@
+    private String path;
+
+    public FileBasedApplicationImportExportManager(APIStore apiStore, String path) {
+        super(apiStore);
+        this.path = path;
+    }
+
+    /**
+     * Export a given Application to a file system as zip archive.
+     * The export root location is given by {@link FileBasedApplicationImportExportManager#path}/exported-application.
+     *
+     * @param application         Application{@link Application} to be exported
+     * @param exportDirectoryName Name of directory to be exported
+     * @return path to the exported directory with exported artifacts
+     * @throws APIMgtEntityImportExportException
+     */
+    public String exportApplication(Application application, String exportDirectoryName) throws
+            APIMgtEntityImportExportException {
+
+        String applicationArtifactBaseDirectoryPath = path + File.separator + exportDirectoryName;
+        try {
+            Files.createDirectories(Paths.get(applicationArtifactBaseDirectoryPath));
+        } catch (IOException e) {
+            String errorMsg = "Unable to create the directory for export Application at :"
+                    + applicationArtifactBaseDirectoryPath;
+            log.error(errorMsg, e);
+            throw new APIMgtEntityImportExportException(errorMsg, e);
+        }
+
+        Application exportApplication = application;
+        String applicationExportDirectory = applicationArtifactBaseDirectoryPath + File.separator +
+                exportApplication.getName();
+        try {
+            //create directory per application
+            Files.createDirectories(Paths.get(applicationExportDirectory));
+            //export application details
+            exportApplicationDetailsToFileSystem(exportApplication,
+                    applicationExportDirectory);
+        } catch (IOException e) {
+            log.error("Error while exporting Application: " + exportApplication.getName(), e);
+        }
+        // Check if no application is exported
+        try {
+            if (APIFileUtils.getDirectoryList(applicationArtifactBaseDirectoryPath).isEmpty()) {
+                // cleanup the archive root directory
+                APIFileUtils.deleteDirectory(path);
+                String errorMsg = "No Applications exported";
+                throw new APIMgtEntityImportExportException(errorMsg, ExceptionCodes.APPLICATION_EXPORT_ERROR);
+            }
+        } catch (APIManagementException e) {
+            String errorMsg = "Unable to find Application Details at: " + applicationArtifactBaseDirectoryPath;
+            throw new APIMgtEntityImportExportException(errorMsg, ExceptionCodes.APPLICATION_IMPORT_ERROR);
+        }
+        return applicationArtifactBaseDirectoryPath;
+    }
+    //create a zip archive of the export directory
+    public String createArchiveFromExportedAppArtifacts(String sourceDirectory, String archiveLocation,
```

Comment:
Shall we add a proper method comment here?

Reply:
Done